freqtrade/freqtradebot.py

```diff
@@ -475,41 +460,65 @@ def execute_buy(self, pair: str, stake_amount: float, price: Optional[float] = N
 
         return True
 
-    def process_maybe_execute_buys(self) -> None:
+    def enter_positions(self) -> int:
         """
-        Tries to execute buy orders for trades in a safe way
+        Tries to execute buy orders for new trades (positions)
         """
-        try:
-            # Create entity and execute trade
-            if not self.create_trades():
-                logger.debug('Found no buy signals for whitelisted currencies. Trying again...')
-        except DependencyException as exception:
-            logger.warning('Unable to create trade: %s', exception)
+        trades_created = 0
 
-    def process_maybe_execute_sells(self, trades: List[Any]) -> None:
+        whitelist = copy.deepcopy(self.active_pair_whitelist)
+        if not whitelist:
+            logger.info("Active pair whitelist is empty.")
+        else:
+            # Remove pairs for currently opened trades from the whitelist
+            for trade in Trade.get_open_trades():
+                if trade.pair in whitelist:
+                    whitelist.remove(trade.pair)
+                    logger.debug('Ignoring %s in pair whitelist', trade.pair)
+
+            if not whitelist:
+                logger.info("No currency pair in active pair whitelist, "
+                            "but checking to sell open trades.")
+            else:
+                # Create entity and execute trade for each pair from whitelist
+                for pair in whitelist:
+                    try:
+                        trades_created += self.create_trade(pair)
+                    except DependencyException as exception:
+                        logger.warning('Unable to create trade: %s', exception)
```

Comment:
I think this message should include the Pair - otherwise it's unclear which pair caused the error as the exception does not necessarily include the pair if it's not relevant for the raising section.

Please review